actors/builtin/market/market_actor.go

```diff
@@ -160,7 +172,7 @@ func (a Actor) PublishStorageDeals(rt Runtime, params *PublishStorageDealsParams
 			amountSlashedTotal = big.Add(amountSlashedTotal, st.updatePendingDealStatesForParty(rt, deal.Proposal.Provider))
 
 			st.lockBalanceOrAbort(rt, deal.Proposal.Client, deal.Proposal.ClientBalanceRequirement())
-			st.lockBalanceOrAbort(rt, deal.Proposal.Provider, deal.Proposal.ProviderBalanceRequirement())
+			st.lockBalanceOrAbort(rt, owner, deal.Proposal.ProviderBalanceRequirement())
```

Comment:
This looks wrong: the collateral should be on balance for for the miner actor, not owner. It _pays out_ to the owner, which is computed at withdrawal time.

Reply:
Still needs fixing.